main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java

```diff
@@ -51,11 +52,17 @@ private String getDownloadURL(String id) throws IOException, NoSuchAlgorithmExce
 			throw new IllegalStateException("No download URL found for track " + id);
 		}
 
-		var downloadInfoLink = json.get("result").values().get(0).get("downloadInfoUrl").text();
-		var downloadInfo = this.sourceManager.getDownloadStrings(downloadInfoLink);
-		if (downloadInfo == null) {
-			throw new IllegalStateException("No download URL found for track " + id);
+		var mp3ItemUrl = json.get("result")
+				.values()
+				.stream()
+				.filter(c -> c.get("codec").text().equals("mp3"))
+				.max(Comparator.comparingInt(b -> b.get("bitrateInKbps").as(Integer.class)))
```

Comment:
in general always use longs
```suggestion
				.max(Comparator.comparingLong(b -> b.get("bitrateInKbps").asLong(0)))
```